pkgs/by-name/mr/mrpack-install/package.nix

```
@@ -0,0 +1,80 @@
+{
+  lib,
+  stdenv,
+  buildGoModule,
+  fetchFromGitHub,
+  buildPackages,
+  installShellFiles,
+  nix-update-script,
+}:
+
+let
+  version = "0.16.10";
+in
+buildGoModule {
+  pname = "mrpack-install";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "nothub";
+    repo = "mrpack-install";
+    tag = "v${version}";
+    hash = "sha256-mTAXFK97t10imdICpg0UI4YLF744oscJqoOIBG5GEkc=";
+  };
+
+  vendorHash = "sha256-az+NpP/hCIq2IfO8Bmn/qG3JVypeDljJ0jWg6yT6hks=";
+
+  ldflags = [
+    "-s"
+    "-w"
```

```suggestion
    "-w"
    "-X github.com/nothub/mrpack-install/buildinfo.version=${version}"
    "-X github.com/nothub/mrpack-install/buildinfo.date=1970-01-01T00:00:00Z"
```

Otherwise you get:
```
> mrpack-install version
mrpack-install ?
```

To verify it works you can add
```
doInstallCheck = true;
nativeInstallCheckInputs = [ versionCheckHook ];
versionCheckProgramArg = [ "version" ];
```

---

Should `buildinfo.date` be set to the current date during buildtime? Or does it simply not matter?

---

It should be set to an arbitrary time, to ensure the build is reproducible.
Unix time 0 is the most common value in nixpkgs.

---

Completely makes sense now that you mention it - will do!